test/integration/controller/jobset_controller_test.go

```diff
@@ -483,27 +486,36 @@ func completeAllJobs(jobList *batchv1.JobList) {
 	}
 }
 
-func completeJob(job *batchv1.Job) {
-	ginkgo.By(fmt.Sprintf("completing job: %s", job.Name))
-	job.Status.Conditions = append(job.Status.Conditions, batchv1.JobCondition{
+func completeJob(stalejob *batchv1.Job) {
+	ginkgo.By(fmt.Sprintf("completing job: %s", stalejob.Name))
+
+	var jobGet batchv1.Job
+	gomega.Eventually(k8sClient.Get(ctx, types.NamespacedName{Name: stalejob.Name, Namespace: stalejob.Namespace}, &jobGet), timeout, interval).Should(gomega.Succeed())
+
+	jobGet.Status.Conditions = append(jobGet.Status.Conditions, batchv1.JobCondition{
 		Type:   batchv1.JobComplete,
 		Status: corev1.ConditionTrue,
 	})
-	gomega.Eventually(k8sClient.Status().Update(ctx, job), timeout, interval).Should(gomega.Succeed())
+	gomega.Eventually(k8sClient.Status().Update(ctx, &jobGet), timeout, interval).Should(gomega.Succeed())
```

Comment:
we need to have the get inside the eventually call so we get the latest version on each retry.

```
gomega.Eventually(func() error {
    var jobGet batchv1.Job
    if err := k8sClient.Get(ctx, types.NamespacedName{Name: stalejob.Name, Namespace: stalejob.Namespace}, &jobGet) {
      return err
    }
    jobGet.Status.Conditions = append(jobGet.Status.Conditions, batchv1.JobCondition{
		Type:   batchv1.JobComplete,
		Status: corev1.ConditionTrue,
     })
    return k8sClient.Status().Update(ctx, &jobGet)}, timeout, interval).Should(gomega.Succeed())
```

Samething with failJob and suspendJob below.

Reply:
applied to those cases also.